sonic-net/sonic-buildimage
# [dhcp_server] Add dhcpservd to dhcp_server container

**Depends on: https://github.com/sonic-net/sonic-buildimage/pull/16548**

#### Why I did it
Part implementation of dhcp_server. HLD: https://github.com/sonic-net/SONiC/pull/1282.
Add dhcpservd to dhcp_server container.

##### Work item tracking
- Microsoft ADO **(number only)**: 25093520

#### How I did it
1. Add installing required pkg (psutil) in `Dockerfile`.
2. Add copying required file to container in `Dockerfile` (kea-dhcp related and dhcpservd related)
3. Add critical_process and supervisor config.
4. Add support for generating kea config (only in dhcpservd.py) and updating lease table (in dhcpservd.py and lease_update.sh)

#### How to verify it
1. Build image with setting `INCLUDE_DHCP_SERVER` to y and enabled dhcp_server feature after installed image, container start as expected.
2. Enter container and found that all processes defined in supervisor configuration running as expected.
3. Kill processes defined in critical_processes, container exist.

unit test
```
========================================== test session starts ==========================================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0
rootdir: /sonic/src/sonic-dhcpservd, configfile: setup.cfg, testpaths: tests
plugins: pyfakefs-5.2.4, cov-2.10.1
collected 26 items                                                                                      

tests/test_dhcp_cfggen.py .........                                                               [ 34%]
tests/test_dhcp_lease.py .....                                                                    [ 53%]
tests/test_dhcp_server_utils.py ........                                                          [ 84%]
tests/test_dhcpservd.py ....                                                                      [100%]Clearing the cache


----------- coverage: platform linux, python 3.9.2-final-0 -----------
Name                             Stmts   Miss Branch BrPart     Cover   Missing
-------------------------------------------------------------------------------
dhcpservd/dhcp_cfggen.py           170     16     72      1    92.98%   49-66, 149-153, 239->241, 241
dhcpservd/dhcp_server_utils.py      37      3     14      0    94.12%   35, 55-56
dhcpservd/dhcp_lease.py             96      3     30      4    94.44%   59->61, 61, 62->63, 63, 75->77, 113->135, 154
dhcpservd/dhcpservd.py              31      1      4      0    97.14%   49
-------------------------------------------------------------------------------
TOTAL                              334     23    120      5    93.83%

Required test coverage of 80.0% reached. Total coverage: 93.83%

========================================== slowest 5 durations ==========================================

```


#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
